src/launchpad/size/insights/android/multiple_native_library_arch.py

```diff
@@ -0,0 +1,65 @@
+from collections import defaultdict
+from pathlib import Path
+
+from launchpad.size.insights.insight import Insight, InsightsInput
+from launchpad.size.models.common import FileInfo
+from launchpad.size.models.insights import (
+    FileSavingsResult,
+    MultipleNativeLibraryArchInsightResult,
+)
+from launchpad.utils.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class MultipleNativeLibraryArchInsight(Insight[MultipleNativeLibraryArchInsightResult]):
+    """Detect APKs that contain multiple native library architectures.
+
+    Most users only need arm64-v8a libraries, but universal APKs often include
+    x86, x86_64, and armeabi-v7a libraries that can be removed.
+    """
+
+    # Architectures that can most likely be removed (keeping only arm64-v8a)
+    REMOVABLE_ARCHITECTURES = {"x86", "x86_64", "armeabi-v7a"}
```

Comment:
Is it more that only one architecture should be shipped in an APK to the app store, rather than non-arm64-v8a archs? Just wondering if generalizing this makes sense, like perhaps a different region like Asia is still on some older architectures.

Reply:
I have only ever seen arm64-v8a in the wild. x86_* are for emulators, and armeabi-v7a is the only remaining. A quick search indicates that this arch is quite outdated at this point, so I think it's a pretty safe bet to keep this as-is